src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -10568,5 +10568,38 @@ public void BaseArguments_Speculation()
 
             Assert.False(model.TryGetSpeculativeSemanticModel(baseWithoutParens.SpanStart + 2, speculativeBase, out _));
         }
+
+        [Fact, WorkItem(54413, "https://github.com/dotnet/roslyn/issues/54413")]
+        public void Repro54413()
+        {
+            var src = @"
+record struct Value(string Text)
+{
+    private Value(int X) { } // 1
+    private Value(Value original) { } // 2
```

Comment:
I assume that before this change, diagnostic '2' and '4' were missing?

Reply:
Diagnostic 2 was missing.
In record classes, copy constructors are special (ie. the concept exists in the spec and has rules around it). That's why we have diagnostic 4, which is different than diagnostic 2.

Reply:
Thanks, I missed that the record below is a record class.￼